src/providers/esmsh.ts

```diff
@@ -4,16 +4,16 @@ import { Resolver } from "../trace/resolver.js";
 import { fetch } from "#fetch";
 import { JspmError } from "../common/err.js";
 
-// The wildcard '*' at the end tells the esm.sh CDN to externalise all
-// dependencies instead of bundling them into the returned module file.
-//   see https://esm.sh/#docs
-const cdnUrl = "https://esm.sh/*";
+const cdnUrl = "https://esm.sh/";
 
 export async function pkgToUrl(pkg: ExactPackage): Promise<`${string}/`> {
-  return `${cdnUrl}${pkg.name}@${pkg.version}/`;
+  // The wildcard '*' at the end tells the esm.sh CDN to externalise all
+  // dependencies instead of bundling them into the returned module file.
+  //   see https://esm.sh/#docs
+  return `${cdnUrl}*${pkg.name}@${pkg.version}/`;
 }
 
-const exactPkgRegEx = /^((?:@[^/\\%@]+\/)?[^./\\%@][^/\\%@]*)@([^\/]+)(\/.*)?$/;
+const exactPkgRegEx = /^(v\d+\/)?\*?((?:@[^/\\%@]+\/)?[^./\\%@][^/\\%@]*)@([^\/]+)(\/.*)?$/;
```

Comment:
just tested the updated regexp, seems it returns incorrect name/version;

<img width="752" alt="Screenshot 2023-05-03 at 19 56 11" src="https://user-images.githubusercontent.com/2883484/235909408-495c2294-7fad-408b-b6f1-5a3a432b3dbb.png">

fix patch:

```js
const [, , name, version] = url.slice(cdnUrl.length).match(exactPkgRegEx) || [];
```